config/generic-mks-sgenl-v2.cfg

```diff
@@ -0,0 +1,171 @@
+## Y Stepper
+##--------------------------------------------------------------------
+[stepper_y]
+step_pin: P0.19
+dir_pin: P0.20
+enable_pin: !P2.8
+rotation_distance: 40
+microsteps: 16
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+endstop_pin: ^P1.27
+position_endstop: -20
+position_min: -20
+position_max: 220
+
+## Make sure to update below for your relevant driver (2208 or 2209)
+[tmc2209 stepper_y]
+uart_pin: P1.8
+run_current: 0.7
+sense_resistor: 0.110
+
+##--------------------------------------------------------------------
+#Z Stepper Settings
+##--------------------------------------------------------------------
+
+## Z0 Stepper
+[stepper_z]
+step_pin: P0.22
+dir_pin: P2.11
+enable_pin: !P0.21
+rotation_distance: 8
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+microsteps: 16
+endstop_pin: ^!P1.28
+position_endstop: 0
+position_min: -1.3
+position_max: 200
+
+
+## Make sure to update below for your relevant driver (2208 or 2209)
+[tmc2209 stepper_z]
+uart_pin: P1.10
+run_current: 0.7
+sense_resistor: 0.110
+
+## Z1 Stepper
+[stepper_z1]
+step_pin: P1.9
+dir_pin: P1.14
+enable_pin: !P0.10
+rotation_distance: 8
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+microsteps: 16
+
+[tmc2209 stepper_z1]
+uart_pin: P1.17
+run_current: 0.1
+sense_resistor: 0.110
+
+#####################################################################
+#Extruder
+#####################################################################
+[extruder]
+step_pin: P2.13
+dir_pin: !P0.11
+enable_pin: !P2.12
+## Update value below when you perform extruder calibration
+## If you ask for 100mm of filament, but in reality it is 98mm:
+## rotation_distance = <previous_rotation_distance> * <actual_extrude_distance> / 100
+rotation_distance: 23.46041055718
+max_extrude_only_distance: 110.0
+microsteps: 16
+full_steps_per_rotation: 200    #200 for 1.8 degree, 400 for 0.9 degree
+nozzle_diameter: 0.400
+filament_diameter: 1.75
+heater_pin: P2.7
+## Validate the following thermistor type to make sure it is correct
+sensor_type: Generic 3950
+sensor_pin: P0.23
+min_temp: 10
+max_temp: 270
+max_power: 1.0
+control = pid
+pid_kp = 20.741
+pid_ki = 0.981
+pid_kd = 109.667
+
+
+[tmc2209 extruder]
+uart_pin: P1.15
+run_current: 0.3
+sense_resistor: 0.110
+#####################################################################
+# Bed Heater
+#####################################################################
+
+[heater_bed]
+heater_pin: P2.5
+sensor_type: Generic 3950
+sensor_pin: P0.24
+control: watermark
+min_temp: 0
+max_temp: 110
+
+
+#####################################################################
+# Fan Control
+#####################################################################
+
+[fan]
+pin: P2.4
+
+[heater_fan hotend_fan]
+pin: P1.4
+heater: extruder
+heater_temp: 60.0
+
+[safe_z_home]
+home_xy_position:10,10
+speed:350
+z_hop:3
```

Comment:
For a generic board cfg, no need to specify [safe_z_home].
Unless (which of course is appreciated) you add a block for bltouch and comment out both blocks

Reply:
I'll comment out safe z home as it's more dependent on the printer 
and add in the bltouch support commented.

Reply:
If you add the bltouch section then a commented [safe_z_home] would be good as for bltouch it is mandatory

Reply:
if bltouch is enabled, would the config compile error out stating that [safe_z_home] is required ?

Reply:
It shall just serve as an example. Comment out both [bltouch] and [safe_z_home] as you already have 👍 
It might be an option to add to the stepper_z, something like
```
endstop_pin: ^!P1.28 # Comment if you use a bltouch
# endstop_pin: probe:z_virtual_endstop # Uncomment if you use a bltouch
```